cobbler/items/profile.py

```diff
@@ -401,14 +401,8 @@ def filename(self, filename: str):
         """
         if not isinstance(filename, str):  # type: ignore
             raise TypeError("Field filename of object profile needs to be of type str!")
-        parent = self.parent
-        if filename == enums.VALUE_INHERITED and parent is None:
+        if filename == enums.VALUE_INHERITED and not self.is_subobject:
```

Comment:
The semantics of the `Item.parent` property has changed, now this property never contains a pointer to an object of another type. Therefore, this change in reality does not change anything.